tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -646,3 +646,141 @@ def _process_resource_set(self, resources, event=None):
                         break
 
         return result
+
+
+class CostFilter(ValueFilter):
+    """
+    Filter resources by the cost consumed over a timeframe.
+    Timeframe can be either number of days before today or one of:
+
+    BillingMonthToDate
+    MonthToDate
+    TheLastBillingMonth
+    TheLastMonth
+    TheLastWeek
+    TheLastYear
+    WeekToDate
+    YearToDate
+
+
+    :examples:
+
+    SQL servers that were cost more than 2000 in the last month.
+
+    .. code-block:: yaml
+
+            policies:
+                - name: expensive sql servers
+                  resource: azure.sqlserver
+                  filters:
+                  - type: cost
+                    timeframe: TheLastMonth
+                    op: gt
+                    value: 2000
+
+    SQL servers that were cost more than 2000 in the last 30 days not including today.
+
+    .. code-block:: yaml
+
+            policies:
+                - name: expensive sql servers
+                  resource: azure.sqlserver
+                  filters:
+                  - type: cost
+                    timeframe: 30
+                    op: gt
+                    value: 2000
+    """
+
+    schema = type_schema('cost',
+        required=['timeframe'],
+        **{
+            'timeframe': {'type': 'string'},
+        },
+        rinherit=ValueFilter.schema)
+
+    schema_alias = True
+
+    preset_timeframes = [
```

Comment:
Can we use this? Rather than redefining it ourselves.
https://github.com/Azure/azure-sdk-for-python/blob/ab599e7c6c371543e48d210c4a11f163951d83d4/sdk/costmanagement/azure-mgmt-costmanagement/azure/mgmt/costmanagement/models/cost_management_client_enums.py